packages/use-storage/src/index.ts

```diff
@@ -0,0 +1,103 @@
+import {
+  useCallback,
+  useEffect,
+  useMemo,
+  useState,
+  useSyncExternalStore,
+} from 'react'
+import { flushSync } from 'react-dom'
+
+declare global {
+  interface WindowEventMap {
+    'event-storage': CustomEvent
+  }
+}
+
+const canUseDOM = !!(
+  typeof window !== 'undefined' &&
+  typeof window?.document !== 'undefined' &&
+  typeof window?.localStorage !== 'undefined' &&
+  typeof window?.sessionStorage !== 'undefined'
+)
+
+const subscribeStorage = (callback: () => void) => {
+  if (canUseDOM) {
+    window.addEventListener('storage', callback)
+    window.addEventListener('event-storage', callback)
```

Comment:
Where does this `event-storage` comes from?

Reply:
From the setValue below which trigger this custom event

Reply:
Why not use the existing `storage` event instead of creating another and listening twice for the same thing?

Reply:
Because `storage` is not broadcasted to current tab :)

https://developer.mozilla.org/en-US/docs/Web/API/Window/storage_event

Reply:
Alright, could you add a comment to explain this?

Reply:
Done :)